[REFACTOR] 섹션 APIs 리팩토링

JSeHoone opened the pull request:
## ❗️ 이슈 번호
#44 

## 📝 작업 내용
1. 임시 유저 정보를 requestParam으로 받아서 구현하였습니다.
2. CustomBusinessException을 사용하여 Service layer에서 에러가 응답되도록 리팩토링 하였습니다.

## 💭 주의 사항

## 💡 리뷰 포인트
- CustomBusinessException 사용 방법이 맞는지 확인 부탁드립니다 !
- user_id로 검사하는 로직이 맞는지 확인부탁드립니다 !

JSeHoone committed:
refactor : Section 생성 시 user 체크

- 임시 user_id를 받아와서 Service layer에서 course를 만든 사람이 맞는지 확인한다.

JSeHoone committed:
refactor : Section 업데이트 시, user 체크

- 임시 user_id를 받아와서 Service layer에서 course를 만든 사람이 맞는지 확인한다.

JSeHoone committed:
refactor : Section 삭제 시, user 체크

- 임시 user_id를 받아와서 Service layer에서 course를 만든 사람이 맞는지 확인한다.

JSeHoone committed:
Merge branch 'develop' into refactor/#44-section-apis-refactoring

JSeHoone committed:
refactor : CustomBusinessException 적용

jjeongdong commented on src/main/java/com/example/projectlxp/section/service/impl/SectionServiceImpl.java:
```diff
@@ -85,12 +101,20 @@ public SectionUpdateResponseDTO modifySection(Long sectionId, String title, int
+        // check who created this section
+        if (findSection.getCourse().getInstructor().getId() != userId) {
+            throw new CustomBusinessException("섹션을 삭제할 권한이 없습니다.", HttpStatus.FORBIDDEN);
+        }
```
공통되는 부분 별도의 메서드로 추출하면 좋을 것 같습니다.

JSeHoone committed:
refactor : Section 권한 체크하는 로직 분리

- SectionValidator를 생성하여 섹션을 관리할 권한이 없으면 throw를 발생시키도록 하였습니다.

- 메서드 오버로딩을 하여 매개변수를 다양하게 받을 수 있도록 만들었습니다.

JSeHoone committed:
refactor : 불필요한 의존성 삭제
